core/src/main/java/com/yahoo/imapnio/async/data/SearchResult.java

```diff
@@ -6,27 +6,50 @@
 import javax.annotation.Nullable;
 
 /**
- * This class provides the list of message sequence numbers from search command response.
+ * This class provides the modification sequence and the list of message sequence numbers from search command response.
  */
 public class SearchResult {
-    /** Search command response sequence number, could be message sequence or UID. */
     @Nonnull
+    /** Search command response sequence number, could be message sequence or UID. */
     private final List<Long> msgNumbers;
 
+    /** Modification sequence, is only shown when CondStore is enabled. */
+    private final Long highestModSeq;
+
     /**
      * Initializes a {@code SearchResult} object with message number collection.
      *
      * @param msgNumbers collection of message number from search command result
      */
     public SearchResult(@Nonnull final List<Long> msgNumbers) {
         this.msgNumbers = msgNumbers;
+        this.highestModSeq = null;
+    }
+
+    /**
+     * Initializes a {@code SearchResult} object with message number collection and modification sequence.
+     *
+     * @param msgNumbers collection of message number from search command result
+     * @param highestModSeq the highest modification sequence from search command result
+     */
+    public SearchResult(@Nonnull final List<Long> msgNumbers, @Nullable final Long highestModSeq) {
```

Comment:
highestModSeq is Nonnull, once you use this constructor, this parameter has to be pass in

Reply:
Actually, for the all results classes, some fields can be null in IMAP response mapper so I use nullable annotation. However, I think if users want to create results without those fields, they should have those options as well.